xbmc/windowing/windows/WinSystemWin32.cpp

```diff
@@ -1304,9 +1304,49 @@ RECT CWinSystemWin32::GetVirtualScreenRect()
   return rect;
 }
 
-int CWinSystemWin32::GetGuiSdrPeakLuminance() const
+/*!
+ * \brief Max luminance for GUI SDR content in HDR mode.
+ * \return Max luminance in nits, lower than 10000.
+*/
+float CWinSystemWin32::GetGuiSdrPeakLuminance()
 {
   const auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
 
-  return settings->GetInt(CSettings::SETTING_VIDEOSCREEN_GUISDRPEAKLUMINANCE);
+  // use cached system value as this is called for each frame
+  if (settings->GetBool(CSettings::SETTING_VIDEOSCREEN_USESYSTEMSDRPEAKLUMINANCE) &&
+      m_validSystemSdrPeakLuminance)
+    return m_systemSdrPeakLuminance;
+
+  // Max nits for 100% UI setting = 1000 nits, < 10000 nits, min 80 nits for 0%
+  int guiSdrPeak = settings->GetInt(CSettings::SETTING_VIDEOSCREEN_GUISDRPEAKLUMINANCE);
+  return 10000.0f / ((100 - guiSdrPeak) * 1.15f + 10);
+}
+
+/*!
+ * \brief Test support of the OS for a SDR max luminance in HDR mode setting
+ * \return true when the OS supports that setting, false otherwise
+*/
+bool CWinSystemWin32::HasSystemSdrPeakLuminance()
```

Comment:
```suggestion
bool CWinSystemWin32::HasSystemSdrPeakLuminance() const
```

Reply:
Not possible without changes not related to the functionality, see comment about same thing in Win10